gemma-core/src/main/java/ubic/gemma/persistence/service/AbstractCriteriaFilteringVoEnabledDao.java

```diff
@@ -16,65 +19,104 @@
  * from a {@link Filters}.
  * @see ubic.gemma.persistence.util.AclCriteriaUtils for utilities to include ACL constraints on the VOs at the
  * database-level.
+ *
+ * @author poirigui
  */
 public abstract class AbstractCriteriaFilteringVoEnabledDao<O extends Identifiable, VO extends IdentifiableValueObject<O>> extends AbstractFilteringVoEnabledDao<O, VO> {
 
-    protected AbstractCriteriaFilteringVoEnabledDao( Class elementClass, SessionFactory sessionFactory ) {
-        super( elementClass, sessionFactory );
-    }
-
-    /**
-     * For the Criterion API, NULL refers to the root entity.
-     *
-     * Override this if you use a custom alias for the root entity.
-     *
-     * @return
-     */
-    @Override
-    public String getObjectAlias() {
-        return null;
+    protected AbstractCriteriaFilteringVoEnabledDao( Class<O> elementClass, SessionFactory sessionFactory ) {
+        // This is a good default objet alias for Hibernate Criteria since null is used to refer to the root entity.
+        super( null, elementClass, sessionFactory );
     }
 
     /**
-     * Obtain a {@link Criteria}.
+     * Obtain a {@link Criteria} for loading VOs.
      *
      * @see ObjectFilterCriteriaUtils#formRestrictionClause(Filters) to obtain a {@link org.hibernate.criterion.DetachedCriteria}
      * from a set of filter clauses.
      */
-    protected abstract Criteria getLoadValueObjectsCriteria( Filters objectFilters );
+    protected abstract Criteria getLoadValueObjectsCriteria( Filters filters );
 
     @Override
-    public Slice<VO> loadValueObjectsPreFilter( Filters objectFilters, Sort sort, int offset, int limit ) {
-        Criteria query = getLoadValueObjectsCriteria( objectFilters );
-        Criteria totalElementsQuery = getLoadValueObjectsCriteria( objectFilters );
+    public Slice<VO> loadValueObjectsPreFilter( Filters filters, Sort sort, int offset, int limit ) {
+        StopWatch stopWatch = StopWatch.createStarted();
+        StopWatch countingStopWatch = StopWatch.create();
+        StopWatch postProcessingStopWatch = StopWatch.create();
+
+        Criteria query = getLoadValueObjectsCriteria( filters );
+        Criteria totalElementsQuery = getLoadValueObjectsCriteria( filters );
+
+        // setup sorting
+        if ( sort != null ) {
```

Comment:
The sorting was missing for some reason. I've added tests covering this in `ExpressionAnalysisResultSetServiceTest`.